Improves handling of ssh error in sync-satellite logs

riya-17 opened the pull request:
Fixes #1609

portante commented on server/bin/pbench-remote-sync-package-tarballs:
```diff
@@ -34,6 +34,8 @@ esac
 
 satellite_config=$1
 output_tar_ball=$2
+tmp=$(dirname $(dirname ${output_tar_ball}))
```
Not sure we can do this, too much knowledge of what the caller is doing in order for this to work safely in all cases.

portante commented on server/bin/pbench-remote-sync-package-tarballs:
```diff
@@ -47,4 +49,4 @@ if [ -z "${remote_opt}" ]; then
 fi
 
-ssh ${remote_host} "${remote_opt}/bin/pbench-sync-package-tarballs" > ${output_tar_ball}
+ssh ${remote_host} "${remote_opt}/bin/pbench-sync-package-tarballs" | tee ${ssh_error} ${output_tar_ball} >/dev/null
```
This is going to break the current method of use for `pbench-remote-sync-package-tarballs`.

The way this works today is by creating a tar ball on the remote side via `tar -cf -`, where the tar ball contents are sent to `stdout` which this script captures to a file given by the caller of `pbench-remote-sync-package-tarballs`.

I think all we have to do is capture `stderr` to a file separately, like:

```
ssh ${remote_host} "${remote_opt}/bin/pbench-sync-package-tarballs" > ${output_tar_ball} 2> ${errors}
```

The `${errors}` file could be specified as a third argument to `pbench-remote-sync-package-tarballs`.

ndokos commented on .gitignore:
```diff
@@ -10,3 +10,9 @@ dist
 .yarnrc
 *.egg-info
 .tox
+pip-wheel-metadata
```
Aren't these changes supposed to be here? Shouldn't they disappear when rebasing on the proper underlying branch (be that master or br69)?

riya-17 replied:
They are here as it has been rebase on br69 but later changed to master. So I had to rebase it again on master

ndokos commented on .travis.yml:
```diff
@@ -7,6 +7,8 @@ env:
   - TOXENV=lint
   - TOXENV=agent
   - TOXENV=server
+  - TOXENV=py3-server
```
Ditto.

riya-17 commented on server/bin/pbench-sync-satellite.sh:
```diff
@@ -119,10 +119,10 @@ typeset -i nfailed_md5=0
 rc=$?
 if [[ $rc != 0 ]] ;then
-    log_exit "$PROG: pbench-remote-sync-package-tarballs: failed." 2
+    log_error "FAILED: $(cat ${tmp}/errors)" "${mail_content}"
```
No nothing specific

riya-17 replied:
@portante log_exit only take numeric argument. To log this message into reports we need ${mail_content} so, I think we should use log_error. What do you say?

portante replied:
Or modify `log_exit` to accept `${mail_content}` as its third argument so it can behave as `log_error` does.

riya-17 replied:
ok

portante commented on server/bin/pbench-base.sh:
```diff
@@ -107,7 +107,12 @@ function log_exit {
+	if [[ -z "${3}" ]]; then
+	    exit ${2}
+    	else
+	    printf -- "%b\n" "${1}" | tee -a "${3}"
```
Indentation problems, tabs vs spaces perhaps?

It seems we might want to consider line 105 as well.  What `log_exit` used to do was send the string, `${PROG}: ${1}` to file descriptor 4.   Now we want that same string also sent to `${3}`, if that argument exists.

So you could consider something like:

```
local _msg="${PROG}: ${1}"
if [[ -z "${3}" ]]; then
    printf -- "%s\n" "${_msg}" >&4
else
    printf -- "%s\n" "${_msg}" | tee -a "${3}" >&4
fi
log_finish
if [[ -z "${2}" ]]; then
    exit 1
else
    exit ${2}
fi
```

portante commented on server/bin/test-bin/ssh:
```diff
@@ -9,5 +9,8 @@ if [ "${remote_host}" = "pbench-satellite.example.com" ] ;then
     # environment from leaking into the "remote" server environment.
     unset _PBENCH_SERVER_CONFIG
     $*
+else
```
I think this mock might be used by other tests eventually.  You might want to check for a specific `${remote_host}` value that would trigger the error.

riya-17 replied:
I did use the value 'pbench-satellite-error.example.com' I assigned that in the config file

portante replied:
Yes, but it would seem you need to check for that value to take this action only when that value is provided.  That way, another test won't get tripped up by that behavior.

riya-17 replied:
ok

portante commented on server/bin/pbench-base.sh:
```diff
@@ -102,12 +102,17 @@ function log_finish {
-    echo "${PROG}: ${1}" >&4
+    local _msg="${PROG}: ${1}"
+    if [[ -z "${3}" ]]; then
+	    printf -- "%s\n" "${_msg}" >&4
```
4 spaces should be used for each level of indentation.

portante commented on server/bin/pbench-base.sh:
```diff
@@ -102,12 +102,17 @@ function log_finish {
+    if [[ -z "${3}" ]]; then
+	    printf -- "%s\n" "${_msg}" >&4
+    else
+	    printf -- "%s\n" "${_msg}" | tee -a "${3}" >&4
```
4 spaces should be used for each level of indentation.

portante commented on server/bin/pbench-base.sh:
```diff
@@ -102,12 +102,17 @@ function log_finish {
     log_finish
     if [[ -z "${2}" ]]; then
-        exit 1
+	    exit 1
```
4 spaces should be used for each level of indentation.

portante commented on server/bin/pbench-base.sh:
```diff
@@ -102,12 +102,17 @@ function log_finish {
+	    exit 1
     else
-        exit ${2}
+	    exit ${2}
```
4 spaces should be used for each level of indentation.

riya-17 commented:
@portante indentation really created a problem, tab gives 8 tabspace while using spaces produced error in flake8, so it was kind of cycle. I have rebased and added added the
